src/core/model/Document.cpp

```diff
@@ -156,14 +156,36 @@ auto Document::createSaveFilename(DocumentType type, const std::string& defaultS
         wildcardString = SaveNameUtils::parseFilenameFromWildcardString(defaultPdfName, this->filepath.filename());
     }
 
-    std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
+
+    std::cout << "Locale has facet: " << std::has_facet<std::codecvt<wchar_t, char, std::mbstate_t>>(std::locale()) << std::endl;
```

Comment:
Consider, that the larger problem is, that the default locale of std::string is std::locale(""), but the strings we have as input is utf-8 (hopefully, otherwise we mix different encodings at other places). So the conversion from utf-8 to wchar_t might fail, since the locale used expects another encoding.
That's why I imbued a locale with utf encoding in my version, but this locale must be installed.

Reply:
The conversion to `wstring`/`wchar_t` could also be omitted, by directly imbuing the utf-8 locale for the time conversion via `ostringstream`. Using wchar_t only has the benefit on Linux, that the code point is UCS4 and **I believe** that they are Unicode characters and that they are the same for all encodings.